packages/mason_logger/lib/src/mason_logger.dart

```diff
@@ -315,7 +315,13 @@ class Logger {
           // show cursor
           ..write('\x1b[?25h')
           ..write('$message ')
-          ..writeln(styleDim.wrap(lightCyan.wrap('$results')));
+          ..writeln(
```

Comment:
Hi @mrverdant13 👋 
Thanks for the PR! Can you please provide a bit more context regarding what the issue was prior to these changes? I personally think `[a, b, c]` is working as expected, thanks!

Reply:
Sure!
Prior to these changes, the provided `display` callback was not applied when showing the selected options.

## Reproduction steps

Given this enum:

```dart
enum Platform {
  android('Android'),
  ios('iOS'),
  web('Web'),
  macos('macOS'),
  windows('Windows'),
  linux('Linux'),
  ;

  const Platform(this.label);

  final String label;
}
```

And this use case:

```dart
final selectedPlatforms = logger.chooseAny<Platform>(
  'Which platforms do you want to support?',
  choices: Platform.values,
  defaultValues: Platform.values,
  display: (platform) => platform.label,
);
```

This is the output:

<img width="700" alt="image" src="https://user-images.githubusercontent.com/42245236/235576811-c4017911-d417-42c3-9e23-e26df8adecb7.png">

As you can see, when selecting the items, the `display` callback is properly used. However, that is not the case when presenting the resulting selected items (in this specific example, they are shown as a stringified list of enum values.)

Reply:
I should have opened an issue before.
My bad 😅

Reply:
Thanks for elaborating! I don't think this is desirable behavior in all cases though. Consider the following:

```dart
enum Shape { square, circle, triangle }
final shape = logger.chooseOne<Shape>(
  'What is your favorite shape?',
  choices: Shape.values,
  display: (shape) => 'Is it a ${shape.name}?',
);
```

I don't think the new behavior makes sense:

```
What is your favorite shape?
❯ ◉  Is it a square?
  ◯  Is it a circle?
  ◯  Is it a triangle?
What is your favorite shape? Is it a square?
```

Reply:
Agreed

Maybe a non-breaking fix would be the use of another nullable callback.
If provided, it would be used to display the resulting selected items. Otherwise, the stringified version of the items would be shown (quite similar to the current behaviour).

```dart
enum Shape {
  square('Awesome Square'),
  circle('Awesome Circle'),
  triangle('Awesome Triangle'),
  ;

  const Shape(this.label);

  final String label;
}
```

## Case 1 - Providing a callback:

```dart
final shape = logger.chooseOne<Shape>(
  'What is your favorite shape?',
  choices: Shape.values,
  display: (shape) => 'Is it a ${shape.name}?',
  resultsDisplay: (shape) => shape.label,
);
```

Output:
```
What is your favorite shape?
❯ ◉  Is it a square?
  ◉  Is it a circle?
  ◯  Is it a triangle?
What is your favorite shape? Awesome Square, Awesome Circle
```

## Case 2 - Without callback:

```dart
final shape = logger.chooseOne<Shape>(
  'What is your favorite shape?',
  choices: Shape.values,
  display: (shape) => 'Is it a ${shape.name}?',
);
```

Output:
```
What is your favorite shape?
❯ ◉  Is it a square?
  ◉  Is it a circle?
  ◯  Is it a triangle?
What is your favorite shape? Shape.square, Shape.circle
```

In both cases, the transformed/readable representation of the results would be joined using `, ` as the separator.

Reply:
I like the proposed solution. `display` implies a generic shape that the options will take in the UI. Given that we have more than one context in which the values could be displayed, dedicated callbacks may make sense.

Reply:
After thinking about this a bit more I actually think the original proposal makes sense. I think it makes sense that the display is used to render the results since that is what the user selected and it's likely just my example that is the outlier. @renancaraujo @wolfenrain are there any cases that you're aware of in which we don't want to render selected results the same way as they are rendered when displayed? I'm tempted to just go forward with @mrverdant13's original proposal.

Reply:
I cant think of any that would fit the criteria, but i do agree with the above that making sure the outlier still is doable/controllable would be favorable.

Reply:
I like the second proposal better, but any of them is better than the current form.

Reply:
Going forward with the original proposal since it's simpler and an improvement over the current behavior. If it's disruptive we can always migrate to a separate API in a subsequent release but I really don't think it's necessary to have a display and results display 🤷